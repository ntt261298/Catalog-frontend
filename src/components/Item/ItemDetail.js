import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { selectItem } from 'utils/selector';
import { getItem } from 'actions/item';
import { errMessage } from 'utils/messages';


export class ItemDetail extends Component {
  state = {
    categoryID: '',
    itemID: '',
  }

  componentDidMount() {
    const { categoryID, itemID } = this.props.params;
    this.props.getItem(categoryID, itemID)
      .catch(err => errMessage(err));
  }

  render() {
    const { item } = this.props;
    return (
      <div className="main">
        <h2>Item Detail</h2>
        <ul>
          { item.map(({ id, title, description }) => (
            <li key={id} className="user-item">
              <h3>{ title }</h3>
              <p>{ description }</p>
            </li>
          )) }
        </ul>
      </div>
    );
  }
}

ItemDetail.propTypes = {
  getItem: PropTypes.func.isRequired,
  item: PropTypes.array.isRequired,
  params: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  item: selectItem(state),
});

export default connect(mapStateToProps, {
  getItem,
})(ItemDetail);
