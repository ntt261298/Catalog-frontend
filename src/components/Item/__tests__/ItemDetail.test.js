import React from 'react';
import { shallow } from '../../../enzyme';
import { ItemDetail } from '../ItemDetail';

describe('src/components/Item/ItemDetail', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(
      <ItemDetail {...props} />,
    );
  };

  beforeEach(() => {
    props = {
      item:
         [
           {
             id: 1,
             items: [],
             name: 'Item 1',
           },
         ],
      params: {
        categoryID: 1,
        itemID: 1,
      },
      getItem: jest.fn().mockImplementation(() => Promise.resolve('message')),
      hideModal: jest.fn(),
      showDeleteItem: jest.fn(),
    };
  });

  it('should render detail Item ', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getItem in componentDidMount', () => {
    setup();
    expect(props.getItem).toHaveBeenCalled();
  });
});
