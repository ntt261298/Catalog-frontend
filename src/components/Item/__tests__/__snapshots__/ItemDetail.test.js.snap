// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Item/ItemDetail should render detail Item  1`] = `
<div
  className="main"
>
  <h2>
    Item Detail
  </h2>
  <ul>
    <li
      className="user-item"
      key="1"
    >
      <h3 />
      <p />
    </li>
  </ul>
</div>
`;
